xbmc/rendering/Extensions.h

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (C) 2017-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "utils/Map.h"
+
+#include <string_view>
+
+struct GLEXTENSIONS
+{
+  enum EXTENSION
+  {
+    APPLE_texture_format_BGRA8888,
+    ARB_multitexture,
+    ARB_pixel_buffer_object,
+    ARB_texture_float,
+    EXT_color_buffer_float,
+    EXT_framebuffer_object,
+    EXT_texture_format_BGRA8888,
+    EXT_unpack_subimage,
+    IMG_texture_format_BGRA8888,
+    KHR_debug,
+    NV_vdpau_interop,
+    NVX_gpu_memory_info,
+    OES_EGL_image_external,
+    EXTENSION_MAX
+  };
+
+  static constexpr auto stringMap = make_map<EXTENSION, std::string_view>({
```

Comment:
Depending on how often this is accessed, wouldn't a `std::unordered_map` make more sense here? You could also use `constexpr std::array<std::string_view, static_cast<std::size_t>` to have this evaluated at compile time.

Reply:
I think it's fine as it is. Yes, the way it is currently used could be achieved with a simple `std::array`, but we don't lose anything by using a `CMap`.

And if at some point in the future this is really used as a map at runtime, we'll be able to make `CMap` lookups n log n with c++20, so we won't lose anything in that scenario either. Especially with these map sizes.

Or was your concern something else and I didn't get it?